File: CS-Examples/09_Charts/CreateParetoChart.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace CreateParetoChart
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file "ParetoChart.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ParetoChart.xlsx")

            ' Get the first worksheet from the workbook
            Dim sheet = workbook.Worksheets(0)

            ' Add a chart to the worksheet
            Dim officeChart = sheet.Charts.Add()

            ' Set the chart type to Pareto
            officeChart.ChartType = ExcelChartType.Pareto

            ' Set the data range for the chart
            officeChart.DataRange = sheet("A2:B8")

            ' Set the position of the chart within the worksheet using row and column indices
            officeChart.TopRow = 1
            officeChart.BottomRow = 19
            officeChart.LeftColumn = 4
            officeChart.RightColumn = 12

            ' Enable binning on the primary category axis (X-axis)
            officeChart.PrimaryCategoryAxis.IsBinningByCategory = True

            ' Set the value to assign to overflow bins
            officeChart.PrimaryCategoryAxis.OverflowBinValue = 5

            ' Set the value to assign to underflow bins
            officeChart.PrimaryCategoryAxis.UnderflowBinValue = 1

            ' Set the color of the Pareto line to blue
            officeChart.Series(0).ParetoLineFormat.LineProperties.Color = Color.Blue

            ' Set the gap width between bars in the chart series
            officeChart.Series(0).DataFormat.Options.GapWidth = 6

            ' Set the chart title to "Expenses"
            officeChart.ChartTitle = "Expenses"

            ' Disable the legend in the chart
            officeChart.HasLegend = False

            ' Save the modified workbook to "Pareto_chart.xlsx"
            workbook.SaveToFile("Pareto_chart.xlsx")
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer("Pareto_chart.xlsx")
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub


	End Class
End Namespace

File: CS-Examples/09_Charts/CreateParetoChart.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace CreateParetoChart
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //Create a new excel document
            Workbook workbook = new Workbook();

            //Load an excel document from the file
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ParetoChart.xlsx");

            // Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            // Add chart
            Chart officeChart = sheet.Charts.Add();

            // Set chart type as Pareto
            officeChart.ChartType = ExcelChartType.Pareto;

            // Set data range in the worksheet
            officeChart.DataRange = sheet["A2:B8"];
            officeChart.TopRow = 1;
            officeChart.BottomRow = 19;
            officeChart.LeftColumn = 4;
            officeChart.RightColumn = 12;
            officeChart.PrimaryCategoryAxis.IsBinningByCategory = true;

            officeChart.PrimaryCategoryAxis.OverflowBinValue = 5;
            officeChart.PrimaryCategoryAxis.UnderflowBinValue = 1;

            // Formatting Pareto line
            officeChart.Series[0].ParetoLineFormat.LineProperties.Color = Color.Blue;

            // Gap width settings
            officeChart.Series[0].DataFormat.Options.GapWidth = 6;

            // Set the chart title
            officeChart.ChartTitle = "Expenses";

            // Hiding the legend
            officeChart.HasLegend = false;

            // Save the workbook to a file
            workbook.SaveToFile("Pareto_chart.xlsx");

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer("Pareto_chart.xlsx");
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }


	}
}
